sdk/communication/Azure.Communication.Common/tests/Identity/CommunicationTokenCredentialTest.cs

```diff
@@ -318,7 +318,33 @@ public void ProactiveRefreshingEnabled_KeepsSchedulingNewTimers()
             Assert.AreEqual(1, testClock.ScheduledActions.Count());
             var firstTimer = testClock.ScheduledActions.First();
             // go into soon to expire window
-            testClock.Tick(TimeSpan.FromMinutes(20 - ThreadSafeRefreshableAccessTokenCache.ProactiveRefreshIntervalInMinutes + 0.5));
+            testClock.Tick(TimeSpan.FromMinutes(20 - CommunicationTokenRefreshOptions.DefaultExpiringOffsetMinutes + 0.5));
+
+            Assert.AreEqual(1, testClock.ScheduledActions.Count());
+            var secondTimer = testClock.ScheduledActions.First();
+            Assert.AreNotEqual(firstTimer, secondTimer);
+        }
+
+        [Test]
+        public void ProactiveRefreshingEnabledWithUserDefinedInterval_KeepsSchedulingNewTimers()
+        {
+            var testClock = new TestClock();
+            var twentyMinToken = GenerateTokenValidForMinutes(testClock.UtcNow, 20);
+            var proactiveRefreshIntervalInMinutes = 10;
+            var refreshOffsetTime = new TimeSpan(0, 0, proactiveRefreshIntervalInMinutes, 0);
+            using var tokenCredential = new AutoRefreshTokenCredential(
+                new CommunicationTokenRefreshOptions(true, refreshOffsetTime, _ => GenerateTokenValidForMinutes(testClock.UtcNow, 20))
+                {
+                    AsyncTokenRefresher = _ => throw new NotImplementedException(),
```

Comment:
Why do you need to replace the default implementation with this one?

Reply:
had  to add test for user-defined auto-refresh interval

Reply:
I meant just this one line. It seems to me that the would work just well without it.

Reply:
in CreateTokenCredentialWithTestClock function initialization of AutoRefreshTokenCredential was implemented this way, so in terms of consistency I just initialized as in the function